src/main/java/org/broadinstitute/dsm/model/rgp/AutomaticProbandDataCreator.java

```diff
@@ -0,0 +1,158 @@
+
+    public static Map<String, List<ParticipantData>> setDefaultProbandDataIfNotExists(Map<String, List<ParticipantData>> participantData,
+                                                                                      Map<String, Map<String, Object>> participantESData,
+                                                                                      @NonNull DDPInstance instance) {
+        if (participantESData == null) {
+            logger.warn("Could not create proband/self data, participant ES data is null");
+            return participantData;
+        }
+        BookmarkDao bookmarkDao = new BookmarkDao();
+        ParticipantDataDao participantDataDao = new ParticipantDataDao();
+        List<FieldSettingsDto> fieldSettingsByOptionAndInstanceId =
+                new FieldSettingsDao().getFieldSettingsByOptionAndInstanceId(Integer.parseInt(instance.getDdpInstanceId()));
+        Map<String, String> columnsWithDefaultOptions =
+                new FieldSettings().getColumnsWithDefaultOptions(fieldSettingsByOptionAndInstanceId);
+        for (Map.Entry<String, Map<String, Object>> entry: participantESData.entrySet()) {
+            String pId = entry.getKey();
+            List<ParticipantData> participantDataList = participantData.get(entry.getKey());
+            Map<String, Object> profile = (Map<String, Object>) entry.getValue().get(ElasticSearchUtil.PROFILE);
+            if (profile == null) {
+                logger.warn("Could not create proband/self data, participant with id: " + pId + " does not have profile in ES");
+                continue;
+            }
+            if (participantDataList == null) {
+                extractAndInsertProbandFromESData(instance, entry, participantDataDao, bookmarkDao, columnsWithDefaultOptions);
+                continue;
+            }
+            boolean isProbandData = participantDataList.stream()
+                    .anyMatch(pData -> (instance.getName().toUpperCase() + NewParticipantData.FIELD_TYPE).equals(pData.getFieldTypeId())
+                        && FamilyMemberConstants.MEMBER_TYPE_SELF.equals(new Gson().fromJson(pData.getData(), Map.class).get(FamilyMemberConstants.MEMBER_TYPE)));
+            if (!isProbandData) {
+                extractAndInsertProbandFromESData(instance, entry, participantDataDao, bookmarkDao, columnsWithDefaultOptions);
+            } else {
+                Optional<ParticipantData> probandData = participantDataList.stream()
+                        .filter(pData -> FamilyMemberConstants.MEMBER_TYPE_SELF.equals(new Gson().fromJson(pData.getData(), Map.class).get(FamilyMemberConstants.MEMBER_TYPE)))
+                        .findFirst();
+                if (probandData.isPresent()) {
+                    updateProbandDataIfESParticipantUpdated(instance, entry, probandData);
+                }
+            }
+        }
+        return ParticipantData.getParticipantData(instance.getName());
```

Comment:
can we change it to only get the participantData again, if something was changed in ddp_participant_data? no need to query the db again if nothing was updated